runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -850,10 +791,6 @@ function LanguageTree:_get_injections()
       self._injection_query:iter_matches(root_node, self._source, start_line, end_line + 1)
     do
       local lang, combined, ranges = self:_get_injection(match, metadata)
-      if not lang then
-        -- TODO(lewis6991): remove after 0.9 (#20434)
-        lang, combined, ranges = self:_get_injection_deprecated(match, metadata)
-      end
```

Comment:
`lang` is now `string?`. Did you not get a warning?

Reply:
I get warnings all the time; hard to see which ones are critical. I have to admit I didn't expect the deprecated fallback to be load bearing here...

So what's the correct fix? Bail if `lang` is `nil`?

Reply:
> I get warnings all the time; hard to see which ones are critical. 

This file should be almost clean with the project `.luarc.json`.

> I have to admit I didn't expect the deprecated fallback to be load bearing here...

Well only because the function will error if it doesn't resolve the match. So in practice it makes little difference, I'm just being strict.

> So what's the correct fix? Bail if lang is nil?

Just don't add the injection if `lang` is `nil`.

Extra points if you can log an error or warning.

Reply:
> Just don't add the injection if lang is nil.

Already done.

> Extra points if you can log an error or warning.

How? `self._log`? What would you like to see? (Feel free to push to this PR if that's easier.)

Reply:
Yeah.

Just some audit that a match from an injection query failed.

Not super important, but I'm gradually putting this kind of messaging in.

Reply:
` self:_log('match from injection query failed')`? what else would you like to see logged here? `match`?

Reply:
pattern index. That will say which injection failed.